specification/cognitiveservices/OpenAI.Inference/models/functions.tsp

```diff
@@ -0,0 +1,64 @@
+import "@typespec/versioning";
+
+namespace Azure.OpenAI;
+
+using TypeSpec.Versioning;
+
+@doc("Specify the name of the only function to call.")
+@added(ServiceApiVersions.v2023_07_01_Preview)
+model FunctionName {
+  @doc("The name of the function to call.")
+  name: string;
+}
+
+@doc("Preset values to control how the mode; responds to function calls.")
+@added(ServiceApiVersions.v2023_07_01_Preview)
+enum FunctionCallPreset {
+  @doc("Means the model can pick between an end-user or calling a function.")
+  auto: "auto",
+
+  @doc("Means the model does not call a function, and responds to the end-user")
+  none: "none",
+}
+
+#suppress "@azure-tools/typespec-autorest/union-unsupported" "External API shape is defined in OpenAPI 3.0 as oneOf."
+alias FunctionCallConfig = FunctionCallPreset | FunctionName;
+
+@doc("""
+  The definition of a function that can be called by the model.
+  See the [guide](/docs/guides/gpt/function-calling) for examples.
+  """)
+@added(ServiceApiVersions.v2023_07_01_Preview)
+model FunctionDefinition {
+  @doc("""
+      The name of the function to be called. Must be a-z, A-Z, 0-9, or contain underscores and dashes,
+      with a maximum length of 64.
+      """)
+  name: string;
+
+  @doc("The description of what the function does.")
+  description?: string;
+
+  #suppress "@azure-tools/typespec-azure-core/no-unknown" "External API shape takes an arbitrary json"
+  @doc("""
+      The parameters the functions accepts, described as a JSON Schema object.
+      See the [guide](/docs/guides/gpt/function-calling) for examples, and the [JSON Schema reference](https://json-schema.org/understanding-json-schema/)
```

Comment:
I am not sure that the `[guide]` link will map to something valid. The prefix of that path is missing.

Reply:
Here is the full link: https://platform.openai.com/docs/guides/gpt/function-calling as per the documentation: https://platform.openai.com/docs/api-reference/chat/create#chat/create-functions

Reply:
I opted to remove the md links and "remarks-style" references entirely for this iteration. We'll likely want to add some additional reference material a bit later on, but I suspect we'd want to link to AOAI documentation (that could in turn link to OpenAI's where appropriate). We'll need to coordinate with the right teams for that part.